### ISSUE TITLE
SHA256 mismatch pepper-flash

### ISSUE THREAD

**Issue by levithomason** [caskroom/homebrew-cask#12671]
![image](https://cloud.githubusercontent.com/assets/5067638/8756879/5ab2faa0-2c8a-11e5-8b3d-cee473ebec3d.png)

Removing the file and running brew update does not work.

**Comment by scribblemaniac** [comment]
I can confirm this behavior. Would you care to submit a PR? It’s a quick fix.

**Comment by levithomason** [comment]
Sure, if you can point me to some docs or info. This will be my first PR for this project.

**Comment by scribblemaniac** [comment]
The *Getting Set Up To Contribute* section of [this document](https://github.com/caskroom/homebrew-cask/blob/master/CONTRIBUTING.md) would be relevant. Rather than making a new cask though, you would just go into your `/Library/Taps/caskroom/homebrew-cask` folder and open up the pepper-flash file with a text editor. Then you just have to replace the expected sha256 from your error with the actual value, also from your error. Then just follow the *Sumitting Your Changes* section of the document I mentioned earlier and you should be good. There isn't too much too it, but there are plenty of people here to help if you need it.

**Comment by adityadalal924** [comment]
Temporary fix submitted as pull request, but there's a better download link on an [Adobe Help](https://helpx.adobe.com/flash-player/kb/installation-problems-flash-player-mac.html) page - [this one](https://fpdownload.macromedia.com/pub/flashplayer/latest/help/install_flash_player_osx_ppapi.dmg)


That link will always point to the latest version, and is an offline installer. However, the uninstall stanza would need to be updated accordingly for this package.

**Comment by Ngrd** [comment]
Thank you @adityadalal924 for the fix and pointing out the URL. However, we prefer the versioned URLs if those are available. At the moment the `pepper-flash.rb` looks good.

**Comment by robsaunders** [comment]
Just got this error today, but with different hashes. I would suspect that adobe just replace the file whenever they feel like it. There seems to be no way to bypass checksum verification...

**Comment by adityadalal924** [comment]
Updated shasum, so that should fix the error for now.

@victorpopkov the "versioned" url is a misnomer in this case, as you can see that `AdobeFlashPlayer_18ppau_a_install.dmg` changes versions silently anyway, breaking the formula with every update.

@robsaunders see my earlier comment - you can create your own custom formula with that url and a version: latest and a shasum of nocheck if you want, or look into a script like [this](https://github.com/rtrouton/rtrouton_scripts/tree/master/rtrouton_scripts/install_latest_adobe_flash_player) one.

**Comment by victorpopkov** [comment]
> the "versioned" url is a misnomer in this case

You are absolutely right, this cask should be modified if it updates silently. There is also a full URL available like `https://admdownload.adobe.com/bin/live/AdobeFlashPlayer_18.0.0.209ppau_a_install.dmg`, but the checksum also seems to change silently and doesn't match with the one used in the cask earlier.

We could consider the approach used in [adobe-air.rb](https://github.com/caskroom/homebrew-cask/blob/807f15d418ecd8d7bfeccd65f0e1317195336b8b/Casks/adobe-air.rb) or [adobe-digital-editions.rb](https://github.com/caskroom/homebrew-cask/blob/807f15d418ecd8d7bfeccd65f0e1317195336b8b/Casks/adobe-digital-editions.rb) and modify the cask like this:
```ruby
cask :v1 => 'pepper-flash' do
  version '18'
  sha256 :no_check # required as upstream package is updated in-place

  url "https://admdownload.adobe.com/bin/live/AdobeFlashPlayer_#{version}ppau_a_install.dmg"
```

**Comment by pochemuto** [comment]
Bug still reproduces

**Comment by tsparber** [comment]
Fixed in  #21196
